### PR TITLE
added socket emit expired token when token has expired. when that is …

### PR DESCRIPTION
**Pull request by stevethomas19**
…emitted the user is logged back out and taken to the login screen.

### CODE REVIEW

File: public/js/admin.js [hunk 1]
```diff
@@ -47,6 +47,11 @@
     $("#messageDisplay").show().one();
   });
 
+  socket.on('expired token', function(service){
+    alert('Your session has expired. Please log in again.');
+    window.location.replace('https://cas.byu.edu/cas/logout');
+    window.location.replace('https://cas.byu.edu/cas/login?service=' + service);
```
Comment (trevren11) [inline public/js/admin.js]: I haven't pulled this down to check but does this wait for the logout to finish then redirect to login?

Comment (stevethomas19) [inline public/js/admin.js]: No. I couldn't figure out a way for it to redirect to the logout page and then to go to the login page. However, it does update the token when they re-login.

Comment (trevren11) [inline public/js/admin.js]: Ok so this just puts me in a forever loop. You might try having it redirect you to the logout and then the user would just go back a page or go back to hlrdesk.

**Comment by stevethomas19** [comment]
Ok I've updated the code again.

Firstly the Alert isn't the browsers alert, its the one we use in hlrdesk and it timeouts after 3 seconds. Then it redirects the user to the logout. From there the user will have to sign back in manually and specify where they want to go in the url.

**Pull request merged**